packages/cli/src/executions/legacy-sqlite-execution-recovery.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Logger } from '@n8n/backend-common';
+import { GlobalConfig } from '@n8n/config';
+import { ExecutionData, ExecutionRepository } from '@n8n/db';
+import { Service } from '@n8n/di';
+
+/**
+ * Service for recovering executions that are missing execution data, this should only happen
+ * for sqlite legacy databases.
+ */
+@Service()
+export class LegacySqliteExecutionRecoveryService {
+	private readonly legacySqlite: boolean;
+
+	constructor(
+		private readonly logger: Logger,
+		private readonly executionRepository: ExecutionRepository,
+		private readonly globalConfig: GlobalConfig,
+	) {
+		const { type: dbType, sqlite: sqliteConfig } = this.globalConfig.database;
+		this.legacySqlite = dbType === 'sqlite' && sqliteConfig.poolSize === 0;
+	}
+
+	/**
+	 * Recover key properties of a truncated execution using event logs.
+	 */
+	async cleanupWorkflowExecutions() {
+		if (!this.legacySqlite) {
+			return;
+		}
+
+		this.logger.info('Starting legacy SQLite execution recovery...');
+
+		const invalidExecutions = await this.executionRepository
```

Comment:
- No context here, what's the exact case that's being reported? An execution has no associated data, or it does have data but it's set to `NULL`? From our discussions I thought it was the latter, but I may be misremembering.
- Should we add tests?

Reply:
I'll add a comment to explain this, but the problem here is that there is no associated data entry available for the execution id.